src/dmd/expressionsem.d

```diff
@@ -5129,12 +5129,40 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             }
             else if (sc.func)
             {
+                FuncDeclaration fd      = s.isFuncDeclaration();
+                AggregateDeclaration ag = s.isAggregateDeclaration();
+                EnumDeclaration ed      = s.isEnumDeclaration();
+                bool inserted     =  (fd || ag || ed || (v && v.isDataseg())) && sc.func.localsymtab.insert(s);
```

Comment:
This is a hot path, and the expression is computed multiple times. Replace with:
```
const bool isFAE = (fd = s.isFuncDeclaration()) !is null ||
                               s.isAggregateDeclaration() !is null ||
                               (ed = s.isEnumDeclaration()) !is null ||
                               (v && v.isDataseg());
```
and ag will never even be used.

Reply:
ag is now used, this was not test before the last force push